### PR TITLE
feat(ratingcontrol): Add Rating Control Styles and Rating Control Sample Page

### LINKED ISSUE #830: [Material] Missing v2 RatingControl Style
RatingControl v2 style should be migrated from v1

### PR DESCRIPTION
**Pull request by nguyeenvy**
closes #830 

﻿GitHub Issue: #830 

## PR Type

What kind of change does this PR introduce?
- Bugfix
- Feature

## Description
Added RatingControl styles in v2/ and RatingControl Sample Page.

Overriden:
image

Overriden with:

	<Page.Resources>
		<SolidColorBrush x:Key="RatingControlForeground"
						Color="Red" />

		<SolidColorBrush x:Key="RatingControlForegroundUnselected"
						Color="Orange" />
		<SolidColorBrush x:Key="RatingControlForegroundSelected"
						Color="Yellow" />
		<SolidColorBrush x:Key="RatingControlPlaceholderForeground"
						Color="Green" />
		<SolidColorBrush x:Key="RatingControlPlaceholderForegroundPointerOver"
						Color="Blue" />
		<SolidColorBrush x:Key="RatingControlForegroundPointerOverUnselected"
						Color="Indigo" />
		<SolidColorBrush x:Key="RatingControlForegroundPointerOverSelected"
						Color="Pink" />
		<SolidColorBrush x:Key="RatingControlForegroundDisabledSelected"
						Color="Red" />

		
		<SolidColorBrush x:Key="SecondaryRatingControlForeground"
						Color="Orange" />

		<SolidColorBrush x:Key="SecondaryRatingControlForegroundUnselected"
						Color="Yellow" />
		<SolidColorBrush x:Key="SecondaryRatingControlForegroundSelected"
						Color="Green" />
		<SolidColorBrush x:Key="SecondaryRatingControlPlaceholderForeground"
						Color="Blue" />
		<SolidColorBrush x:Key="SecondaryRatingControlPlaceholderForegroundPointerOver"
						Color="Indigo" />
		<SolidColorBrush x:Key="SecondaryRatingControlForegroundPointerOverUnselected"
						Color="Pink" />
		<SolidColorBrush x:Key="SecondaryRatingControlForegroundPointerOverSelected"
						Color="Red" />
		<SolidColorBrush x:Key="SecondaryRatingControlForegroundDisabledSelected"
						Color="Orange" />
		<SolidColorBrush x:Key="RatingControlCaptionForeground"
						Color="Pink" />
		<SolidColorBrush x:Key="SecondaryRatingControlCaptionForeground"
						 Color="Green" />
	</Page.Resources>

## PR Checklist 
- [X] Tested UWP
- [ ] Tested iOS
- [ ] Tested Android
- [X] Tested WASM
- [ ] Tested MacOS

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+                                <VisualState x:Name="PointerOverSet">
+                                    <VisualState.Setters>
+                                        <Setter Target="ForegroundContentPresenter.Foreground" Value="{ThemeResource RatingControlForegroundSelected}" />
+                                    </VisualState.Setters>
+                                </VisualState>
+                            </VisualStateGroup>
+                        </VisualStateManager.VisualStateGroups>
+
+                        <StackPanel Grid.Row="0"
+                                    Margin="-20,-20,-20,-20"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:131]: can you remove the negative margin + positive nested margin pairs
to see if this weird hack was even needed?

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                                (There's 4px of whitespace accounted for in the text area in the redline)
+                            -->
+                            <!-- TODO MSFT: 9925444 Fix vertical alignment in all text scenarios -->
+                        </StackPanel>
+
+                        <ContentPresenter x:Name="ForegroundContentPresenter"
+                                          Grid.Row="0"
+                                          IsHitTestVisible="False">
+                            <!-- Margin is on the StackPanel because ContentPresenter clips differently such that moving the reverse margin up won't work -->
+                            <StackPanel Margin="-40,-40,-40,-40"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:163]: same here -margin and nested +margin

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml (hunk 1 above)
Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:131]: this also seem to ruin the vertical alignment of label vs stars

**Review by Xiaoy312 — COMMENTED** [review]
in visual studio, can you go to: tools\options\XamlStyler\Attribute Formatting\Attribute Indentation Style
and change this value to `Mixed`?
and format the xaml files again

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml (hunk 1 above)
Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:131]: weird, these seem to come straight from: https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/RatingControl/RatingControl.xaml

Comment (nguyeenvy) [reply src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:131]: @Xiaoy312 When I remove the negative and positive margins, the Label is being cut off, so some part of it doesn't show. I am trying to play around with the margins to see if anything can fix that cut in the Label without having negative margins.

Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:131]: in that case, i guess we will just keep it as it...

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+            <StaticResource x:Key="RatingControlCaptionStyle" ResourceKey="CaptionMedium" />
+            <StaticResource x:Key="RatingControlFontFamily" ResourceKey="SymbolThemeFontFamily" />
+
+            <StaticResource x:Key="SecondaryRatingControlCaptionForeground" ResourceKey="OnSurfaceBrush" />
+            <StaticResource x:Key="SecondaryRatingControlCaptionFontFamily" ResourceKey="MaterialRegularFontFamily" />
+            <StaticResource x:Key="SecondaryRatingControlCaptionStyle" ResourceKey="CaptionMedium" />
+        </ResourceDictionary>
+    </ResourceDictionary.ThemeDictionaries>
+
+    <Style x:Key="DefaultRatingControlStyle"
```
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:86]:
```suggestion
    <Style x:Key="MaterialRatingControlStyle"
```

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+                                            IsHitTestVisible="False"
+                                            Orientation="Horizontal" />
+                            </StackPanel>
+                        </ContentPresenter>
+                    </Grid>
+                </ControlTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+</ResourceDictionary>
```
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:270]:
```suggestion
	<Style x:Key="MaterialDefaultRatingControlStyle"
		   TargetType="muxc:RatingControl"
		   BasedOn="{StaticResource MaterialRatingControlStyle}" />
</ResourceDictionary>
```

@nguyeenvy missing the default style (to match the coming changes https://github.com/unoplatform/Uno.Themes/pull/1044/files)

File: src/library/Uno.Material/Styles/Controls/v2/_Resources.xaml [hunk 5]
```diff
@@ -33,6 +33,7 @@
 	<Style BasedOn="{StaticResource MaterialProgressBarStyle}" TargetType="muxc:ProgressBar" />
 	<Style BasedOn="{StaticResource MaterialProgressRingStyle}" TargetType="muxc:ProgressRing" />
 	<Style BasedOn="{StaticResource MaterialRadioButtonStyle}" TargetType="RadioButton" />
+	<Style BasedOn="{StaticResource MaterialRatingControlStyle}" TargetType="RatingControl" />
```
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/_Resources.xaml:36]:
```suggestion
	<Style BasedOn="{StaticResource MaterialDefaultRatingControlStyle}" TargetType="RatingControl" />
```

File: src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+                        </ContentPresenter>
+                    </Grid>
+                </ControlTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+
+    <Style x:Key="SecondaryRatingControlStyle"
+           TargetType="muxc:RatingControl"
+           BasedOn="{StaticResource DefaultRatingControlStyle}">
```
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/RatingControl.xaml:183]:
```suggestion
           BasedOn="{StaticResource MaterialRatingControlStyle}">
```

File: src/samples/UWP/Uno.Themes.Samples.Shared/Content/Controls/RatingControlSamplePage.xaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            </sample:SamplePageLayout.MaterialTemplate>
+            <sample:SamplePageLayout.M3MaterialTemplate>
+                <DataTemplate>
+                    <StackPanel>
+                        <!-- Default style -->
+                        <TextBlock Text="Default RatingControl style with label"
+                                   Style="{StaticResource TitleSmall}"
+                                   Margin="0,0,16,0" />
+                        <smtx:XamlDisplay UniqueKey="M3_Material_DefaultRatingControlSamplePage">
+                            <muxc:RatingControl Style="{StaticResource DefaultRatingControlStyle}"
```
Comment (agneszitte) [inline src/samples/UWP/Uno.Themes.Samples.Shared/Content/Controls/RatingControlSamplePage.xaml:46]:
```suggestion
                            <muxc:RatingControl Style="{StaticResource MaterialRatingControlStyle}"
```

**Review by agneszitte — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

**Comment by agneszitte** [comment]
https://github.com/Mergifyio backport legacy/2x

**Pull request merged**